### PR TITLE
feat(composer): support pasting and dragging images to upload

### PR DESCRIPTION
**Pull request by jtgi**
## Changes
- adds pasting and drag support for images in the composer
- fixes editing profile image: right now it always says 'file type not supported'

## Notes
- The imgur npm package doesn't work for me. Trying to change my profile pic on production gives the same  'unsupported filetype' error (they're valid filetypes). The library seems dated; switching to standard http form data worked.
- I hacked a local dev setup as I couldn't find anything in the readme, it could use a good pass from @hellno or whomever.

## Video
https://github.com/user-attachments/assets/8f90cbbc-5877-4dc0-a822-3aead92781ec

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -93,7 +93,6 @@
     "focus-trap-react": "^10.2.1",
     "fuse.js": "^7.0.0",
     "google-protobuf": "^3.21.2",
-    "imgur": "^2.4.2",
```
Comment (jtgi) [inline package.json:96]: this library doesn't work – its used in the edit profile page only and on prod it's broken with a 'unsupported file type' error. Switched to axios / standard http, works fine.

File: pages/_app.tsx [hunk 2]
```diff
@@ -93,12 +93,7 @@ export default function MyApp({ Component, pageProps }: AppProps) {
   );
 
   return (
-    <ThemeProvider
-      attribute="class"
-      defaultTheme="light"
-      enableSystem
-      disableTransitionOnChange
-    >
+    <ThemeProvider attribute="class" defaultTheme="light" enableSystem disableTransitionOnChange>
```
Comment (jtgi) [inline pages/_app.tsx:96]: sadly my line length for prettier is wider – feel free to reformat or maybe there's .vscode workspace settings that should be standardized

File: src/common/components/Editor/NewCastEditor.tsx [hunk 3]
```diff
@@ -160,55 +153,86 @@ export default function NewPostEntry({
...
-    "meta+enter",
-    onSubmitPost,
-    [onSubmitPost, draft, account],
-    {
-      enableOnFormTags: true,
+  const ref = useHotkeys("meta+enter", onSubmitPost, [onSubmitPost, draft, account], {
+    enableOnFormTags: true,
+  });
+
+  const { uploadImage, isUploading, error, image } = useImgurUpload();
```
Comment (jtgi) [inline src/common/components/Editor/NewCastEditor.tsx:160]: reusable hook for here and edit profile

File: src/common/components/Editor/NewCastEditor.tsx [hunk 4]
```diff
@@ -160,55 +153,86 @@ export default function NewPostEntry({
...
+      renderChannelsSuggestionConfig: createRenderMentionsSuggestionConfig({
+        getResults: getChannels,
+        RenderList: ChannelList,
+      }),
+      renderMentionsSuggestionConfig: createRenderMentionsSuggestionConfig({
+        getResults: getMentions,
+        RenderList: MentionList,
+      }),
+      editorOptions: {
+        editorProps: {
```
Comment (jtgi) [inline src/common/components/Editor/NewCastEditor.tsx:212]: this is the integration point for both paste and drag with tiptap/prosemirror

File: package.json (hunk 1 above)
Comment (hellno) [reply package.json:96]: I love removing packages 🙌🏼

File: pages/_app.tsx (hunk 2 above)
Comment (hellno) [reply pages/_app.tsx:96]: no worries, we have an upcoming PR to standardize this :)

**Review by hellno — APPROVED** [review]
LGTM 🥳🥳 really cool contribution, thank you!

**Pull request merged**